### PR TITLE
WIP gRPC client

### PR DESCRIPTION
**Pull request by JamesNK**
Early WIP

### CODE REVIEW

**Comment by sebastienros** [comment]
Looks good. Took most of it from the SignalR worker?

File: src/BenchmarksClient/Workers/GrpcWorker.cs [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+        {
+            if (_stopped)
+            {
+                return;
+            }
+
+            _requestsPerConnection[channelId] += 1;
+
+            var latency = end - start;
+            latency = latency.Add(TimeSpan.FromMilliseconds(_clientToServerOffset));
```
Comment (BrennanConroy) [inline src/BenchmarksClient/Workers/GrpcWorker.cs]: Remove this

File: src/BenchmarksClient/Workers/GrpcWorker.cs [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+                            {
+                                Log($"{id}: Starting {_scenario}");
+
+                                var client = new Greeter.GreeterClient(_channels[id]);
+
+                                while (!cts.IsCancellationRequested)
+                                {
+                                    try
+                                    {
+                                        var start = DateTime.UtcNow;
```
Comment (BrennanConroy) [inline src/BenchmarksClient/Workers/GrpcWorker.cs]: stop watch might be better for you here

File: src/BenchmarksClient/Workers/GrpcWorker.cs [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+            }
+            catch (Exception ex)
+            {
+                var text = "Exception from test: " + ex.Message;
+                Log(text);
+                _job.Error += Environment.NewLine + text;
+            }
+
+            Log($"Waiting on duration");
+            cts.Token.WaitHandle.WaitOne();
```
Comment (BrennanConroy) [inline src/BenchmarksClient/Workers/GrpcWorker.cs]: Consider using something like https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/common/testassets/Tests.Utils/CancellationTokenExtensions.cs

File: src/BenchmarksClient/Workers/GrpcWorker.cs [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+            foreach (var channel in _channels)
+            {
+                tasks.Add(channel.ShutdownAsync());
+            }
+
+            await Task.WhenAll(tasks);
+            Log("Channels have been disposed");
+
+            // TODO: Remove when clients no longer take a long time to "cool down"
+            await Task.Delay(5000);
```
Comment (BrennanConroy) [inline src/BenchmarksClient/Workers/GrpcWorker.cs]: You can probably remove this for grpc

**Pull request merged**